File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.java
import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        Set<String> phoneSet = new HashSet<>();
        for (String number: phone_book) {
            phoneSet.add(number);
        }
        
        for (String number: phone_book) {
            StringBuilder temp = new StringBuilder();
            for (int i = 0; i < number.length()-1; i++) {
                temp.append(number.charAt(i));
                if (phoneSet.contains(temp.toString())) {
                    return false;
                }
            } 
        }
        
        return true;
    }
}

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    hash_map = {}
    
    for number in phone_book:
        hash_map[number] = 1
        
    for number in phone_book:
        temp = ""
        for n in number:
            temp += n
            if temp in hash_map and temp != number:
                return False
    return True